core/src/main/java/com/datastax/oss/driver/internal/core/os/JnrNativeImpl.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.os;
+
+import com.datastax.oss.driver.internal.core.util.Reflection;
+import java.lang.reflect.Method;
+import java.util.Optional;
+import java.util.function.Supplier;
+import jnr.ffi.Platform;
+import jnr.posix.POSIX;
+import jnr.posix.POSIXFactory;
+import jnr.posix.Timeval;
+import jnr.posix.util.DefaultPOSIXHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JnrNativeImpl implements NativeImpl {
+
+  private static final Logger LOG = LoggerFactory.getLogger(JnrNativeImpl.class);
+
+  private final Optional<POSIX> posix;
+  private final Optional<Platform> platform;
+
+  public JnrNativeImpl() {
+
+    this.posix = loadPosix();
+    this.platform = loadPlatform();
+  }
+
+  private Optional<POSIX> loadPosix() {
+
+    try {
+      return Optional.of(POSIXFactory.getPOSIX(new DefaultPOSIXHandler(), true))
+          .flatMap(this::validatePosix);
+    } catch (Throwable t) {
+      LOG.debug("Error loading POSIX", t);
+      return Optional.empty();
+    }
+  }
+
+  private Optional<POSIX> validatePosix(POSIX posix) {
+
+    try {
+
+      posix.getpid();
+    } catch (Throwable t) {
+
+      LOG.debug("Error calling getpid()", t);
+      return Optional.empty();
+    }
+
+    try {
+
+      Timeval tv = posix.allocateTimeval();
+      int rv = posix.gettimeofday(tv);
+      if (rv != 0) {
+
+        LOG.debug("Expected getitimeofday() to return zero, observed {}", rv);
+        return Optional.empty();
+      }
+    } catch (Throwable t) {
+
+      LOG.debug("Error calling gettimeofday()", t);
+      return Optional.empty();
+    }
+
+    return Optional.of(posix);
+  }
```

Comment:
There's a _slight_ difference in behaviour from the old impl here.  We've moved both calls into the jnr-posix orbit but don't maintain discrete vals for the validation of each discrete call; we just validate the whole thing by confirming that we can call _both_ native functions.  Logically this means that we could find ourselves in a situation where getpid() succeeded but gettimeofday() didn't, and in that case we'd prevent users from subsequent calls to _either_ syscall when (in theory) we could allow them to call getpid() and just throw on gettimeofday() calls.

Note that gettimeofday() "failing" above means throwing an exception (i.e.a failure in the Java infrastructure) not just a non-zero return val (which indicates that the syscall was made and something went wrong within the kernel).

I kinda like this idea even in the abstract; if we can't validate that your POSIX instance can do _everything_ we expect it's in an untrustworthy state and we don't allow you to do anything.  More practically, it seems _extremely_ unlikely that we'd get a Java exception with one syscall but not another; the whole point of POSIX is that we expect syscalls with these names and params to be available.  If one is there and another isn't we've got some kind of very odd libc we shouldn't trust much anyway.

Reply:
> More practically, it seems extremely unlikely that we'd get a Java exception with one syscall but not another

💯 
I'm fine with the change.